src/infra/auth/auth.ts

```diff
@@ -22,6 +23,16 @@ export const userHasPermission = (
 	return user.permissions.includes(permission);
 };
 
+let getNewUserRole: (user: AnyObject) => string = () => 'default-user';
+
+export const setRegistrationRoleFunc = (
+	registerFunc: (user: AnyObject) => string,
+) => {
+	getNewUserRole = registerFunc;
+};
+
+export const getRegistrationRole = (user: AnyObject) => getNewUserRole(user);
```

Comment:
I think you can inherit types like:
```suggestion
export const getRegistrationRole: typeof getNewUserRole = (user) => getNewUserRole(user);
```